src/clj/game/core/actions.clj

```diff
@@ -250,20 +247,14 @@
                               (Exception. "Error in a card-title prompt") 25)))
           (.println *err* (str "Current prompt: " prompt))))
 
-      ;; Default text prompt
+      ;; Otherwise, choices is a sequence of strings and/or cards
+      ;; choice is a string and should match one of the strings, or the title of one of the cards
       :else
-      (let [buttons (filter #(or (= choice %)
-                                 (same-card? card %)
-                                 (let [choice-str (if (string? choice)
-                                                    (lower-case choice)
-                                                    (lower-case (:title choice "do-not-match")))]
-                                   (or (= choice-str (lower-case %))
-                                       (= choice-str (lower-case (:title % ""))))))
-                            choices)
-            button (first buttons)]
-        (if button
-          (do (when-let [effect-prompt (:effect prompt)]
-                (effect-prompt button))
+      (let [match (first (filter #(or (= choice %)
+                                       (= choice (:title % "")))
+                                  choices))]
+        (if match
+          (do ((:effect prompt) match)
```

Comment:
Not all prompts will have effects, so you'll want to leave the `when-let` in.

Reply:
Huh. If there's no effect in the default case does the prompt just do nothing?

Reply:
You're right, I was thinking of a different part of the engine. Please ignore me.